### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by coolreader18**

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for creating a TypeScript definition. 🙌 Can you follow [this styleguide](https://github.com/sindresorhus/typescript-definition-style-guide)?

**Comment by coolreader18** [comment]
To follow that, should I put `exports.default = export;` in `index.js`?

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Split a string on the first occurrence of a given separator.
+ * 
+ * @param string - The string to split.
+ * @param separator - The separator to split on.
+ * @returns The string split on the separator, if possible.
+ */
+declare function splitOnFirst(string: string, separator: string): [string, string?];
```
Comment (sindresorhus) [inline index.d.ts:8]: You don't need to declare. You can just export it directly.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Split a string on the first occurrence of a given separator.
+ * 
+ * @param string - The string to split.
+ * @param separator - The separator to split on.
+ * @returns The string split on the separator, if possible.
```
Comment (sindresorhus) [inline index.d.ts:6]: This is unclear. It makes it seem it returns the string. Should be clearer.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Ping :)

**Comment by sindresorhus** [comment]
Closing in favor of #4

**Pull request closed without merging**